features/netsocket/SocketAddress.cpp

```diff
@@ -290,6 +290,8 @@ bool operator==(const SocketAddress &a, const SocketAddress &b)
     } else if (a._addr.version == NSAPI_IPv6) {
         return memcmp(a._addr.bytes, b._addr.bytes, NSAPI_IPv6_BYTES) == 0;
     }
+
+    MBED_UNREACHABLE;
```

Comment:
Nit - why not just return false here rather than "while (1)" (from MBED_UNREACHABLE)?